Ambiguous occurrence ‘quotRemInteger’ when building with ghc-9.0

with ghc-9.0.1,
```
Blaze/Text/Int.hs:117:30: error:
    Ambiguous occurrence ‘quotRemInteger’
    It could refer to
       either ‘GHC.Num.quotRemInteger’,
              imported from ‘GHC.Num’ at Blaze/Text/Int.hs:26:17-30
           or ‘GHC.Integer.GMP.Internals.quotRemInteger’,
              imported from ‘GHC.Integer.GMP.Internals’ at Blaze/Text/Int.hs:30:1-32
              (and originally defined in ‘GHC.Integer’)
    |
117 |     splith p (n:ns) = case n `quotRemInteger` p of
    |                              ^^^^^^^^^^^^^^^^
```
Cf. "Big-number support" in https://downloads.haskell.org/~ghc/9.0.1/docs/html/users_guide/9.0.1-notes.html#highlights - which I can't quite make sense of. I am trying
```
#if defined(INTEGER_GMP)
-- import GHC.Integer.GMP.Internals
#elif defined(INTEGER_SIMPLE)
import GHC.Integer.Simple.Internals
#endif
```
but I am getting 
```
Blaze/Text/Int.hs:104:10: error:
    Not in scope: data constructor ‘S#’
    Perhaps you meant ‘I#’ (imported from GHC.Types)
    |
104 | integer (S# i#) = int (I# i#)
    |          ^^
```
Ok then, this makes it compile (and tests pass)
```
integer :: Integer -> Builder
#if defined(INTEGER_GMP)
-- integer (S# i#) = int (I# i#)
#endif
```
but it certainly looks like this should be fixed differently (handling flags in the cabal file)

--- a/Blaze/Text/Int.hs
+++ b/Blaze/Text/Int.hs
@@ -23,7 +23,10 @@ import Data.Int (Int8, Int16, Int32, Int64)
 import Data.Monoid (mappend, mempty)
 import Data.Word (Word, Word8, Word16, Word32, Word64)
 import GHC.Base (quotInt, remInt)
+#if MIN_VERSION_base(4,15,0)
+#elif
 import GHC.Num (quotRemInteger)
+#endif
 import GHC.Types (Int(..))
 
 #if defined(INTEGER_GMP)

